### PR TITLE
fix(dedicated): cannot delete BYOIP

### PR DESCRIPTION
**Pull request by vks-ovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/sprint-2236-2238`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-9449
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~~[ ] Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

Cannot delete byoip from IP page, so moved the delete functionality for BYOIP to service page and removed delete option from IP page.

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html [hunk 1]
```diff
@@ -306,7 +306,9 @@
         </li>
 
         <!-- IP DELETE -->
-        <li data-ng-if="ipBlock.type === 'FAILOVER' && ipBlock.isUniq">
+        <li
+            data-ng-if="ipBlock.type === 'FAILOVER' && ipBlock.isUniq && !ipBlock.bringYourOwnIp"
+        >
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html:311]: Can you call a readable function here.

File: packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js [hunk 2]
```diff
@@ -38,4 +37,32 @@ export default class TerminateServiceCtrl {
         this.loading = false;
       });
   }
+
+  terminateByoipService() {
+    return this.BillingTerminate.terminateByoipService(this.serviceId)
+      .then(() =>
```
Comment (sachinrameshn) [inline packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js:43]: then, catch and finally block seems duplicate of terminateService call. Can we avoid duplication ?

Comment (vks-ovh) [reply packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js]: @sachinrameshn  moved to common function. Thanks.

File: packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html (hunk 1 above)
Comment (vks-ovh) [reply packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html]: taken care.

File: packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js [hunk 3]
```diff
@@ -16,26 +16,45 @@ export default class TerminateServiceCtrl {
...
+    return this.BillingTerminate.terminateByoipService(this.serviceId)
+      .then(() => this.success())
+      .catch((err) => this.error(err))
       .finally(() => {
         this.loading = false;
       });
   }
+
+  terminate() {
+    this.loading = true;
```
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js:39]: **nitpick:** setting the `loading` flag here and unsetting it in different functions causes confusions and it is not readable.

File: packages/manager/apps/dedicated/client/app/ip/ip/ip-ip.controller.js [hunk 4]
```diff
@@ -679,4 +679,10 @@ export default /* @ngInject */
   };
 
   refreshAlerts();
+
+  $scope.canByoipDeleted = function canByoipDeleted(ipBlock) {
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/ip/ip-ip.controller.js:683]:
```suggestion
  $scope.canDeleteByoip = function canDeleteByoip(ipBlock) {
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/ip/ip/ip-ip.controller.js (hunk 4 above)
Comment (vks-ovh) [reply packages/manager/apps/dedicated/client/app/ip/ip/ip-ip.controller.js]: taken care.

File: packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js (hunk 3 above)
Comment (vks-ovh) [reply packages/manager/modules/billing-components/src/components/cancellation-form/billing-confirmTerminate.controller.js]: taken care.

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html [hunk 5]
```diff
@@ -306,7 +306,7 @@
         </li>
 
         <!-- IP DELETE -->
-        <li data-ng-if="ipBlock.type === 'FAILOVER' && ipBlock.isUniq">
+        <li data-ng-if="canByoipDeleted(ipBlock)">
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/ip/popover/ip.html:309]:
```suggestion
        <li data-ng-if="canDeleteByoip(ipBlock)">
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report test-checkout-marie#4.10 ✘  
* 
  * New Job ✘

**Pull request closed without merging**